src/evaluator.ts

```diff
@@ -166,34 +171,38 @@ export class Evaluator {
                     });
                 },
                 error => {
-                    const errorResponse: ErrorResponse = {
-                        title: formatMessage(error),
-                        type: EvaluationErrorType.UNEXPECTED_ERROR,
-                        detail: 'Please try again or contact Croct support if the error persists.',
-                        status: 500, // Internal Server Error
-                    };
-
-                    reject(new EvaluationError(errorResponse));
+                    if (!abortController.signal.aborted) {
+                        reject(
+                            new EvaluationError({
+                                title: formatMessage(error),
+                                type: EvaluationErrorType.UNEXPECTED_ERROR,
+                                detail: 'Please try again or contact Croct support if the error persists.',
+                                status: 500, // Internal Server Error
+                            }),
+                        );
+                    }
                 },
```

Comment:
Why are we using `onReject` callback instead of `.catch`? Catching seems more adequate.

Reply:
They are not the same. Calling `catch` after the `then` would chain then instead of adding them to the same settling point.

```ts
// This promise will reject with `TypeError: Cannot read properties of null`
Promise.resolve(null)
  .then(
    (value: string | null) => value!.toUpperCase(),
    // This onRejected has no relation with the onResolved above
    error => console.log(error),
  );


// This promise resolves to undefined and logs the error `TypeError: Cannot read properties of null`
Promise.resolve(null)
  .then((value: string | null) => value!.toUpperCase())
  // This catch will also capture errors from the above then
  .catch(error => console.log(error));
```

Reply:
Another way of visualizing is translating that to `async`/`await` syntax:

```ts
const promise = Promise.resolve(null);

// Using onRejected
let value: string | null;
try {
  value = await promise;
} catch (error) {
  return console.log(error)
}
// Error happens outside of the try-catch
return value!.toUpperCase();

// Chaining them together
try {
  const value: string | null = await promise;
  // Error happens inside of the try-catch
  return value!.toUpperCase();
} catch (error) {
  return console.log(error)
}
```